apps/src/templates/census2017/CensusMap.jsx

```diff
@@ -0,0 +1,345 @@
+    schoolName: PropTypes.string.isRequired,
+    city: PropTypes.string.isRequired,
+    state: PropTypes.string.isRequired,
+    location: PropTypes.string.isRequired,
+    teachesCs: PropTypes.string.isRequired
+  };
+
+  render() {
+    let censusMessage;
+
+    switch (this.props.teachesCs) {
+      case 'YES':
+        censusMessage = "We believe this school offers Computer Science.";
+        break;
+      case 'NO':
+        censusMessage = "We believe this school offers limited or no Computer Science opportunities.";
+        break;
+      case 'MAYBE':
+        censusMessage = "We have conflicting data for this school.";
+        break;
+      default:
+        censusMessage = "We need data for this school.";
+    }
+
+    const schoolDropdownOption = {
+      value: this.props.schoolId,
+      label: `${this.props.schoolName} - ${this.props.city}, ${this.props.state}`,
+      school: {
+        nces_id: this.props.schoolId,
+        name: this.props.schoolName,
+        city: this.props.city,
+        state: this.props.state,
+        latitude: this.props.location.split(',')[0],
+        longitude: this.props.location.split(',')[1],
+      },
+    };
+
+    return (
+      <div className="census-info-window">
+        <h4>
+          <b>
+            {this.props.schoolName}
+          </b>
+          <br />
+          ({this.props.city}, {this.props.state})
+        </h4>
+        <hr />
+        <div className="census-message">{censusMessage}</div>
+        <div className="button-container">
+          <div className="button-link-div">
+            <a onClick={() => this.props.handleTakeSurveyClick(schoolDropdownOption)}>
+              <div className="button">
+                <div className="button-text">Take the survey for this school</div>
+              </div>
+            </a>
+          </div>
+          <div className="button-link-div">
+            <a href="/yourschool/letter" target="_blank">
+              <div className="button">
+                <div className="button-text">Send the survey to a teacher at this school</div>
+              </div>
+            </a>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default class CensusMap extends Component {
+  static propTypes = {
+    onSchoolChange: PropTypes.func.isRequired,
+    fusionTableId: PropTypes.string.isRequired,
+    school: PropTypes.object,
+  };
+
+  gmap;
+  scrollwheelOption = false;
+  draggableOption = false;
+  resizeThrottleTimerId;
+  resizeThrottleTimeoutMs = 500;
+  maxHeightPercentage = 0.65;
+  infoWindow;
+
+  componentWillMount() {
+    this.infoWindow = new google.maps.InfoWindow();
+  }
+
+  componentDidMount = () => {
+    this.initializeMap();
+    this.resizeMap();
+
+    $(window).resize(() => {
+      // Throttle calling of resizeMap
+      clearTimeout(this.resizeThrottleTimerId);
+      this.resizeThrottleTimerId = setTimeout(this.resizeMap, this.resizeThrottleTimeoutMs);
```

Comment:
Technically you're doing a debounce here, not a throttle.  Throttle behavior might be preferable.  Possible future cleanup: Use [`_.throttle()`](https://lodash.com/docs/4.17.5#throttle).

Reply:
Noted. This code was pulled over from the haml.